### PR TITLE
Tooling feed updates to support .NET 9

### LINKED ISSUE #550: Tooling feed updates to support .NET 9

### PR DESCRIPTION
**Pull request by surgupta-msft**
resolves #550 

Tooling feed updates to support .NET 9

Diff between 4.84.0 and 4.85.0 - https://gist.github.com/surgupta-msft/737c4f7048bc5df898ca0bfaa5ee253c/revisions

Diff between net8 and net9 - https://gist.github.com/surgupta-msft/0f978c67ee04aa2cf0cc9dfd0b27a851/revisions

### CODE REVIEW

File: cli-feed-v4.json [hunk 1]
```diff
@@ -27055,6 +27055,412 @@
...
+      "workerRuntimes": {
+        "dotnet": {
+          "net6": {
+            "displayInfo": {
+              "displayName": ".NET 6.0",
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET",
+              "description": "LTS",
+              "endOfLifeDate": "2024-11-12T00:00:00Z"
```
Comment (surgupta-msft) [inline cli-feed-v4.json:27070]: @mattchenderson from where we can get this `endOfLifeDate` ?

File: cli-feed-v4.json [hunk 2]
```diff
@@ -27055,6 +27055,412 @@
...
+            }
+          },
+          "net9-isolated": {
+            "displayInfo": {
+              "displayName": ".NET 9.0",
+              "hidden": true,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 9",
+              "description": "Isolated",
+              "endOfLifeDate": "2026-11-12T00:00:00Z"
```
Comment (surgupta-msft) [inline cli-feed-v4.json]: @mattchenderson from where we can get this `endOfLifeDate` ?

File: cli-feed-v4.json [hunk 3]
```diff
@@ -27055,6 +27055,412 @@
...
+              "netFrameworkVersion": "v8.0"
+            },
+            "linuxSiteConfig": {
+              "linuxFxVersion": "DOTNET-ISOLATED|8.0"
+            }
+          },
+          "net9-isolated": {
+            "displayInfo": {
+              "displayName": ".NET 9.0",
+              "hidden": true,
```
Comment (surgupta-msft) [inline cli-feed-v4.json]: @mattchenderson Not sure if the value of hidden should be `true` or `false` for Net9?

File: cli-feed-v4.json [hunk 4]
```diff
@@ -27055,6 +27055,412 @@
...
+              "displayVersion": "v4",
+              "targetFramework": ".NET 9",
+              "description": "Isolated",
+              "endOfLifeDate": "2026-11-12T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8,net9",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "1.16.2"
```
Comment (surgupta-msft) [inline cli-feed-v4.json]: @kshyju Do we update the latest worker sdk version here? or should we do that in a separate PR to keep things independent?

File: cli-feed-v4.json [hunk 5]
```diff
@@ -27055,6 +27055,412 @@
...
+            "default": false,
+            "toolingSuffix": "net9-isolated",
+            "localEntryPoint": "dotnet.exe",
+            "targetFramework": "net9.0",
+            "itemTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ItemTemplates.NetCore/4.0.3041",
+            "projectTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ProjectTemplates/4.0.3038",
+            "projectTemplateId": {
+              "csharp": "Microsoft.AzureFunctions.ProjectTemplate.CSharp.Isolated.3.x"
+            },
+            "localContainerBaseImage": "DOCKER|mcr.microsoft.com/azure-functions/dotnet-isolated:4-dotnet-isolated8.0-appservice",
```
Comment (surgupta-msft) [inline cli-feed-v4.json]: @kshyju @soninaren from where can we get the value of `localContainerBaseImage` for Net9?

File: cli-feed-v4.json [hunk 6]
```diff
@@ -27055,6 +27055,412 @@
...
+              "FUNCTIONS_WORKER_RUNTIME": "dotnet"
+            },
+            "windowsSiteConfig": {
+              "netFrameworkVersion": "v6.0"
+            },
+            "linuxSiteConfig": {
+              "linuxFxVersion": "DOTNET|6.0"
+            }
+          },
+          "net5-isolated": {
```
Comment (kshyju) [inline cli-feed-v4.json]: Is it safe to remove "net5-isolated" ?

File: cli-feed-v4.json (hunk 5 above)
Comment (kshyju) [reply cli-feed-v4.json]: We need to build it. There is a GH item to track that work.

File: cli-feed-v4.json [hunk 7]
```diff
@@ -56,7 +56,7 @@
       "hidden": false
     },
     "v0-prerelease": {
-      "release": "4.83.0-inprocess",
+      "release": "4.84.0-inprocess",
```
Comment (kshyju) [inline cli-feed-v4.json]: Do we need to update this?

File: cli-feed-v4.json (hunk 5 above)
Comment (kshyju) [reply cli-feed-v4.json]: https://github.com/Azure/azure-functions-dotnet-worker/issues/2590

File: cli-feed-v4.json (hunk 6 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: I am not sure. I can check with the team. If we decide to remove it, I would prefer to do that in a separate PR.

File: cli-feed-v4.json (hunk 7 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: Yeah we don't need it. Reverted.

File: cli-feed-v4.json [hunk 8]
```diff
@@ -27055,6 +27055,315 @@
...
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "1.16.2"
+            },
+            "default": false,
+            "toolingSuffix": "net9-isolated",
+            "localEntryPoint": "dotnet.exe",
+            "targetFramework": "net9.0",
+            "itemTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ItemTemplates.NetCore/4.0.3041",
+            "projectTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ProjectTemplates/4.0.3038",
```
Comment (kshyju) [inline cli-feed-v4.json]: We need to use the new project template with the net9 change.

Once [this PR](https://github.com/Azure/azure-functions-templates/pull/1542) merges, we need to publish the nuget package and use that here.

File: cli-feed-v4.json (hunk 2 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: Taken from here - https://github.com/dotnet/core/blob/main/releases.md

File: cli-feed-v4.json (hunk 8 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: Let me know once the templates are published, I will update here. Or I can also do that in the next iteration of tooling feed changes - when I will be updating coretools and localContainerBaseImage versions.

File: cli-feed-v4.json [hunk 9]
```diff
@@ -46,9 +46,9 @@
       "hidden": false
     },
     "v4-prerelease": {
-      "release": "4.83.0",
+      "release": "4.84.0",
       "releaseQuality": "Prerelease",
-      "hidden": true
+      "hidden": false
```
Comment (mattchenderson) [inline cli-feed-v4.json:51]: The hidden flag should not change for the tag.

File: cli-feed-v4.json [hunk 10]
```diff
@@ -27055,6 +27055,315 @@
...
+              "linuxFxVersion": "DOTNET-ISOLATED|8.0"
+            }
+          },
+          "net9-isolated": {
+            "displayInfo": {
+              "displayName": ".NET 9.0",
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 9",
+              "description": "Isolated",
```
Comment (mattchenderson) [inline cli-feed-v4.json]:
```suggestion
              "description": "Isolated Preview",
```

File: cli-feed-v4.json [hunk 11]
```diff
@@ -27055,6 +27055,315 @@
...
+              "endOfLifeDate": "2026-05-12T00:00:00Z"
```
Comment (mattchenderson) [inline cli-feed-v4.json]: Is there a citation for this? As far as I know, an end of life date for .NET 9 has not yet been set because it's a preview version. I think this line needs to be removed.

File: cli-feed-v4.json [hunk 12]
```diff
@@ -27055,6 +27055,315 @@
...
+            "default": false,
+            "toolingSuffix": "net9-isolated",
+            "localEntryPoint": "dotnet.exe",
+            "targetFramework": "net9.0",
+            "itemTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ItemTemplates.NetCore/4.0.3041",
+            "projectTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ProjectTemplates/4.0.3038",
+            "projectTemplateId": {
+              "csharp": "Microsoft.AzureFunctions.ProjectTemplate.CSharp.Isolated.3.x"
+            },
+            "localContainerBaseImage": "DOCKER|mcr.microsoft.com/azure-functions/dotnet-isolated:4-dotnet-isolated8.0-appservice",
```
Comment (mattchenderson) [inline cli-feed-v4.json]: this will need to change as well - we can line it up even if that tag isn't published yet.
```suggestion
            "localContainerBaseImage": "DOCKER|mcr.microsoft.com/azure-functions/dotnet-isolated:4-dotnet-isolated9.0-appservice",
```

File: cli-feed-v4.json (hunk 3 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: It should be `true` for now since we have not yet released support. We can do a separate PR to unhide it.

**Review by mattchenderson — DISMISSED** [review]

File: cli-feed-v4.json [hunk 13]
```diff
@@ -27172,7 +27172,7 @@
               "displayVersion": "v4",
               "targetFramework": ".NET 7",
               "description": "Isolated",
-              "endOfLifeDate": "2024-05-10T00:00:00Z"
+              "endOfLifeDate": "2024-05-11T00:00:00Z"
```
Comment (mattchenderson) [inline cli-feed-v4.json]: If we're updating this here, .NET 7 end-of-support was May 14th per https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

It would also be great to get .NET 8 updated, as we seem to have the wrong date for it, too. It's correct for in-proc, but isolated has it 2 days late.

File: cli-feed-v4.json (hunk 11 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: I took it from here - https://github.com/dotnet/core/blob/main/releases.md. Let me know if we should still remove this line.

File: cli-feed-v4.json (hunk 9 above)
Comment (surgupta-msft) [reply cli-feed-v4.json:51]: My bad. I was just using it to test locally. Resolved it.

File: cli-feed-v4.json (hunk 8 above)
Comment (kshyju) [reply cli-feed-v4.json]: Please use 4.0.3062

File: cli-feed-v4.json (hunk 6 above)
Comment (kshyju) [reply cli-feed-v4.json]: .NET 5 is no longer supported, so I recommend removing it. Since these tag entries are immutable and cannot be updated once added, it's a good idea to check-in the correct set of changes from the start if possible. This approach will help minimize the file size.

Comment (surgupta-msft) [reply cli-feed-v4.json]: yes, thats correct. I will update here.

**Comment by surgupta-msft** [comment]
Few things have been identified to come up in the follow-up PR -

- Unhide Net9-isolated
- Add `endOfLifeDate` once the date is available for Net9 GA
- Update worker.sdk version with Net9 support once the package is GA
- Update coretools links when [these changes](https://github.com/Azure/azure-functions-core-tools/pull/3764) are released

**Review by mattchenderson — COMMENTED** [review]
changes seem good overall - my only question is if net5-isolated needs to be removed at this time. I'd probably rather see that be a separate set of changes if we're going to be doing that, unless it is required as part of getting net9 out.

**Review by mattchenderson — CHANGES_REQUESTED** [review]
Changing the positioning back to "request changes" to ensure .NET 5 is not removed with this PR.

**Review by fabiocav — COMMENTED** [review]
Agree with @mattchenderson 's comment about .NET 5 here. 

It's something we should be doing, but needs coordination and shouldn't be combined in this PR

File: cli-feed-v4.json (hunk 6 above)
Comment (surgupta-msft) [reply cli-feed-v4.json]: Update - not removing net5-isolated with this PR as per Fabio's and Matthew's comments below - https://github.com/Azure/azure-functions-tooling-feed/pull/549#pullrequestreview-2219442244

**Review by mattchenderson — APPROVED** [review]

File: cli-feed-v4.json [hunk 14]
```diff
@@ -27429,6 +27429,314 @@
...
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "1.16.2"
+            },
+            "default": false,
+            "toolingSuffix": "net9-isolated",
+            "localEntryPoint": "dotnet.exe",
+            "targetFramework": "net9.0",
+            "itemTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ItemTemplates.NetCore/4.0.3062",
+            "projectTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.Functions.Worker.ProjectTemplates/4.0.3062",
```
Comment (kshyju) [inline cli-feed-v4.json:27632]: Let's update to the latest version for both of these packages ( 4.0.5006.)

**Review by kshyju — APPROVED** [review]
